generator/src/update-dependencies.ts

```diff
@@ -0,0 +1,88 @@
+/**********************************************************************
+ * Copyright (c) 2018-2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+/**
+ * Simple code to rewrite versions of a dependency in package.json files
+ * @author Thomas Mäder
+ */
+
+import * as glob from 'glob';
+import * as path from 'path';
+
+import { CommandBuilder } from 'yargs';
+import { promisify } from 'util';
+import { rewriteJson } from './json-utils';
+
+export const builder: CommandBuilder = {
+    'theia-version': {
+        describe: 'The version of Theia to uses in che-theia',
+        requiresArg: true,
+        type: 'string',
+        demandOption: true,
+    },
+    'che-theia': {
+        describe: 'Path of the che-theia project source',
+        requiresArg: true,
+        type: 'string',
+        demandOption: false,
+    },
+};
+
+const EXCLUSIONS = ['@theia/plugin-packager'];
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function handleCommand(args: any) {
+    const cheTheiaDir = args['che-theia '] || process.cwd();
+    const theiaVersion = args['theia-version'];
+
+    const packageJsons = await promisify(glob)(cheTheiaDir + '/extensions/*/package.json');
+    packageJsons.push(...(await promisify(glob)(cheTheiaDir + '/plugins/*/package.json')));
+    packageJsons.push(path.resolve(cheTheiaDir, 'generator/package.json'));
+    packageJsons.push(path.resolve(cheTheiaDir, 'assembly/package.json'));
+    packageJsons.push(path.resolve(cheTheiaDir, 'package.json'));
+
+    for (const packageJson of packageJsons) {
+        await updateDependencies(packageJson, packageName => {
+            if (packageName.startsWith('@theia') && !EXCLUSIONS.includes(packageName)) {
+                return theiaVersion;
+            }
+        });
+    }
+}
+
+async function updateDependencies(
+    packageJSONPath: string,
+    replaceVersion: (packageName: string) => string | undefined
+) {
+    try {
+        await rewriteJson(packageJSONPath, pkgJson => {
+            // we're assuming the package.json is well formed
+            replaceInSection(pkgJson.dependencies, replaceVersion);
+            replaceInSection(pkgJson.devDependencies, replaceVersion);
+        });
+    } catch (e) {
+        console.warn('could not find package json: ' + packageJSONPath);
```

Comment:
```suggestion
        console.warn(`could not find package json: ${packageJSONPath}`);
```

Reply:
It really doesn't matter. What makes you think string interpolation is intrinsically better in this case?

Reply:
https://rules.sonarsource.com/typescript/RSPEC-3512

Reply:
So there's no real reason, it's just the opinion of some person at sonarsource that it's clearer? I disagree 🤷

Reply:
if people run analysis tools on repositories, they'll look only at number of "code smell" reports and won't look deeply if it is related to Thomas agreement or not.

Reply:
As far as I know, there is no requirement to heed sonarsource warnings on the Eclipse che-theia project. Was there a decision by the development team I'm not aware of? @ericwill maybe you have more info?

Reply:
For a print statement/warning I would say it's a question of code style as nothing is concretely impacted by doing it one or the other (in this case). I personally would have opted for the template string but it's not a hill I think we need to die on.

@benoitf if you think we should have sonarlint run on PRs then let's open an issue for that and discuss it there